lib/hashtable.cc

```diff
@@ -120,6 +121,82 @@ void Hashtable::consume_fasta(
     consume_fasta<SeqIO>(parser, total_reads, n_consumed);
 }
 
+void Hashtable::_check_batches(unsigned int num_batches, unsigned int batch)
+{
+    bool powerof2 = !(num_batches == 0) && !(num_batches & (num_batches - 1));
+    if (!powerof2) {
+        std::string message = "num_batches must be a power of 2";
+        throw InvalidValue(message);
+    }
+    if (batch >= num_batches) {
+        std::string message = "batch must be less than num_batches";
+        throw InvalidValue(message);
+    }
+}
+
+template<typename SeqIO>
+void Hashtable::consume_fasta_banding(
+    std::string const &filename, unsigned int num_batches, unsigned int batch,
+    unsigned int &total_reads, unsigned long long &n_consumed
+)
+{
+    _check_batches(num_batches, batch);
+    ReadParserPtr<SeqIO> parser = get_parser<SeqIO>(filename);
+    consume_fasta_banding<SeqIO>(parser, num_batches, batch, total_reads,
+                                 n_consumed);
+}
+
+template<typename SeqIO>
+void Hashtable::consume_fasta_banding(
+    ReadParserPtr<SeqIO>& parser, unsigned int num_batches, unsigned int batch,
+    unsigned int &total_reads, unsigned long long &n_consumed
+)
+{
+    _check_batches(num_batches, batch);
+
+    Read read;
+    while (!parser->is_complete()) {
+        try {
+            read = parser->get_next_read();
+        } catch (NoMoreReadsAvailable) {
+            break;
+        }
+
+        if (read.sequence.length() < _ksize) {
+            continue;
+        }
+
+        bool is_valid = true;
```

Comment:
Two concerns / observations.

- Comparing cleaned seq vs original seq will disqualify reads with lowercase ACGT, which I think is a lousy default with unexpected consequences.
- The current `consume_fasta` function in Hashtable doesn't use this utility yet either.

As @ctb mentioned, there's some work to be done with removing redundant read cleaning code (see #1590), and that's not all going to be resolved in this PR. Ultimately this will be handled by the parser instead of the consume function, right?